goldilocks/src/poseidon2.rs

```diff
@@ -7,14 +7,24 @@
 //! to build the 4t x 4t matrix used for the external (full) rounds).
 
 //! Long term we will use more optimised internal and external linear layers.
+use alloc::vec::Vec;
 
 use p3_poseidon2::{
     external_final_permute_state, external_initial_permute_state, internal_permute_state,
-    matmul_internal, ExternalLayer, HLMDSMat4, InternalLayer, MDSMat4, NoPackedImplementation,
+    matmul_internal, ExternalLayer, ExternalLayerConstructor, HLMDSMat4, InternalLayer,
+    InternalLayerConstructor, MDSMat4, Poseidon2,
 };
 
 use crate::{to_goldilocks_array, Goldilocks};
 
+pub type Poseidon2Goldilocks<const WIDTH: usize> = Poseidon2<
+    Goldilocks,
+    Poseidon2ExternalLayerGoldilocks<WIDTH>,
+    Poseidon2InternalLayerGoldilocks,
+    WIDTH,
+    7,
```

Comment:
Not strictly part of this PR, but same comment here for `7` as for `5` in the M31 file: It should be given a name like `const SBOX_DEGREE: usize = 7`.

(It is really a property of the field, so specifying it separately is a bit strange, but I guess it's to allow the use of a more expensive s-box than necessary if someone wants?)

Reply:
Hmm yeah I might leave that for now but this is something we should come back to. There is (in some cases) a possible advantage to using a more expensive s-box as it changes the number of internal/external rounds you need. But I'm pretty sure it's never optimal.